plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -74,4 +345,20 @@ private List<Long> getSnapshotsIdsByCreationOrder(String tableName)
                 .map(row -> (Long) row.getField(ID_FIELD))
                 .collect(toList());
     }
+
+    private List<MaterializedRow> getSnapshotSummariesByCreationOrder(String tableName)
+    {
+        return getQueryRunner().execute(
+                format("SELECT CAST(SUMMARY as JSON) FROM \"%s$snapshots\" ORDER BY committed_at", tableName))
+                .getMaterializedRows();
+    }
+
+    private List<QueryId> getQueryIdsFromSnapshotSummariesByCreationOrder(String tableName)
```

Comment:
why list?
Shouldn't we check the last snapshot's query id?

Reply:
Listing was beneficial when I was first making the larger end to end test, as well as just for debugging when making the tests

It _did_ allow me to see that one part added two snapshots (which I believe is expected behavior - same query ID from the transaction).

But I think I can just get the latest now provided I remove the larger end to end test.

Reply:
Added a new function that simply returns the latest query id.

If we're good with that, I can remove this old function if we get rid of the end 2 end test (arguably it's not needed now that things are broken up).